api/ws/workers/peersUpdateRules.js

```diff
@@ -0,0 +1,120 @@
+'use strict';
+
+var connectionsTable = require('./connectionsTable');
+var Z_schema = require('../../../helpers/z_schema');
+var schema = require('../../../schema/transport');
+
+var self, z_schema =  new Z_schema();
+
+/**
+ * Secures peers updates. Used only by workers.
+ * @param {SlaveWAMPServer} slaveWAMPServer - used to send verified update requests to master process.
+ * @constructor
+ */
+function PeersUpdateRules (slaveWAMPServer) {
+	this.slaveWAMPServer = slaveWAMPServer;
+	self = this;
+}
+
+/**
+ * Sends requests to main process with SocketCluster authKey attached
+ * @param {string} procedureName
+ * @param {Object} peer
+ * @param {function} cb
+ */
+PeersUpdateRules.prototype.sendInternally = function (procedureName, peer, cb) {
+	this.slaveWAMPServer.sendToMaster(procedureName, {peer: peer, authKey: this.slaveWAMPServer.worker.options.authKey}, peer.nonce, cb);
+};
+
+PeersUpdateRules.prototype.internal = {
+
+	/**
+	 * @throws {Error} if peer does not exist
+	 * @throws {Error} if peer.nonce or connectionId is undefined/null/0
+	 * @throws {Error} if peer registered connection before
+	 * @throws {Error} if peer doesn't have nonce, height or broadhash fields
+	 * @throws {Error} if error has been returned from main process
+	 * @param {Object} peer
+	 * @param {string} connectionId
+	 * @param {function} cb
+	 */
+	insert: function (peer, connectionId, cb) {
+		if (!peer || !peer.nonce) {
+			return cb('Cannot insert peer without nonce');
+		}
+		if (connectionsTable.getConnectionId(peer.nonce)) {
+			return cb('Peer with nonce ' + peer.nonce + ' is already inserted');
+		}
+		if (connectionsTable.connectionIdToNonceMap[connectionId]) {
+			return cb('Connection id ' + connectionId + ' is already assigned');
+		}
+		try {
+			connectionsTable.add(peer.nonce, connectionId);
+			self.sendInternally('acceptPeer', peer, function (err) {
+				if (err) {
+					connectionsTable.remove(peer.nonce);
+				}
+				return cb(err);
+			});
+		} catch (ex) {
+			return cb(ex);
```

Comment:
are you sure that `ex` will be always there? maybe `return cb(ex || 'Some error');`